agent/agent.go

```diff
@@ -1072,9 +1067,34 @@ LOAD:
 	return nil
 }
 
+// RegisterEndpoint registers a handler for the consul RPC server
+// under a unique name while making it accessible under the provided
+// name. This allows overwriting handlers for the golang net/rpc
+// service which does not allow this.
+func (a *Agent) RegisterEndpoint(name string, handler interface{}) error {
```

Comment:
That is correct. The only challenge is that the DNS test is also using it and I've stumbled across this when I tried moving the DNS server into a separate package. Might be good to not expose this to make sure I won't forget it during the DNS refactor